features/events/api/src/main/java/org/opennms/netmgt/events/api/EventParameterUtils.java

```diff
@@ -157,4 +161,20 @@ public static String format(Parm parm) {
     public static Map<String, Parm> normalize(final List<Parm> parmList) {
         return parmList.stream().collect(Collectors.toMap(Parm::getParmName, Function.identity(), (p1, p2) -> p2));
     }
+
+    public static List<Parm> normalizePreserveOrder(final List<Parm> parmList) {
+        Objects.requireNonNull(parmList);
+        Set<String> existingNames = new HashSet<>();
+        LinkedList<Parm> filteredList = new LinkedList<>();
+        // go through the list backwards since we want to keep the last element of duplicates names
+        // (existing behaviour)
```

Comment:
@j-white do we want to preserve the existing behavior (keep the last element of duplicates names). If not we can simplify this method and pick the first duplicate parameter instead of the last.

Reply:
Let's preserve the existing behavior.

Reply:
ok.